File: lib/misc.py
import subprocess
from os import chdir
from lib.utils import sys_ok, sys_err
from lib.colors import RED, RESET

def clear_command():
    try:
        subprocess.run(["clear"], check=True)
        print(sys_ok("This is a funny clear command :D"))
    except subprocess.CalledProcessError:
        pass

def update_command():
    try:
        chdir('/tmp')
        subprocess.run([
            "bash", "-c",
            "sudo rm -rf pck3r && sudo git clone https://github.com/amzy31/pck3r pck3r && cd pck3r && sudo make install"
        ], check=True)
        print(sys_ok("pck3r updated successfully!"))
    except subprocess.CalledProcessError:
        print(f"{sys_err('')}{RED}Failed to update pck3r.{RESET}")

def version_command():
    print(f"{sys_ok('')}version : 0.2")

def print_help():
    from lib.utils import print_help as ph
    ph()

File: lib/misc.js
const GLib = imports.gi.GLib;
const { sys_ok, sys_err } = imports.lib.utils;
const { RED, RESET } = imports.lib.colors;

this.clear_command = function() {
    try {
        const [, , stderr, exit_status] = GLib.spawn_command_line_sync('clear');
        if (exit_status === 0) {
            print(sys_ok('This is a funny clear command :D'));
        }
    } catch (e) {
        // pass
    }
};

this.update_command = function() {
    try {
        GLib.chdir('/tmp');
        const [, , stderr, exit_status] = GLib.spawn_command_line_sync(
            'bash -c "sudo rm -rf pck3r && sudo git clone https://github.com/amzy31/pck3r pck3r && cd pck3r && sudo make install"'
        );
        if (exit_status === 0) {
            print(sys_ok('pck3r updated successfully!'));
        } else {
            print(`${sys_err('')}${RED}Failed to update pck3r.${RESET}`);
        }
    } catch (e) {
        print(`${sys_err('')}${RED}Failed to update pck3r.${RESET}`);
    }
};

this.version_command = function() {
    print(`${sys_ok('')}version : 1.0`);
};


